firmware/thruna/BMS/Src/App/App_Accumulator.c

```diff
@@ -436,13 +437,30 @@ bool App_Accumulator_CheckFaults(struct Accumulator *const accumulator, struct T
     float max_allowable_cell_temp = MAX_CELL_DISCHARGE_TEMP_DEGC;
     float min_allowable_cell_temp = MIN_CELL_DISCHARGE_TEMP_DEGC;
 
+    float cell_voltage_minimum = 0;
+    float cell_voltage_sum;
+
+    for(uint8_t seg; seg <= 4; seg++){
+        float cell_voltage = App_Accumulator_GetAverageCellVoltage(accumulator, seg);
+        cell_voltage_sum += cell_voltage;
+        if(cell_voltage < cell_voltage_minimum){
+            cell_voltage_minimum = cell_voltage;
+        }
+    }
+    float cell_voltage_avg = cell_voltage_sum/4.0;
+    float segment_current = App_TractiveSystem_GetCurrent(ts);
+
+    float num_of_cells = (segment_current)*INTERNAL_CELL_RESISTANCE_OHMS/
+                        (cell_voltage_avg - cell_voltage_minimum + segment_current*INTERNAL_CELL_RESISTANCE_OHMS/3.0);
```

Comment:
Hmm I don't think `cell_voltage_minimum` and `cell_voltage_avg` are what we want. 

It looks like `cell_voltage_minimum` is the minimum of the average cell voltages for each segment - you want the individual cell with the lowest voltage. This info is in the `accumulator->voltage_status.min_voltage` variable.

`cell_voltage_avg` is `cell_voltage_sum` divided by 4? where does this come from?

Reply:
`num_of_cells` isn't super descriptive - maybe `estimated_min_cells_in_parallel`?

Reply:
Also this math isn't super obvious - maybe link to the confluence doc?

Reply:
Also also can you add a check to make sure we don't actually divide by zero? I don't think its possible but I don't want the board to crash if that ever does magically happen...